File: variant-x/csharp-service/Models/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FlashSale.Api.Models;

public enum OrderStatus
{
    Pending,
    Confirmed,
    Processing,
    Shipped,
    Delivered,
    Cancelled,
    Refunded
}

public enum PaymentStatus
{
    Pending,
    Authorized,
    Captured,
    Failed,
    Cancelled,
    Refunded
}

/// <summary>
/// Order model for both regular and flash sale orders
/// </summary>
[Table("orders")]
public class Order : BaseEntity
{
    [Required]
    [StringLength(50)]
    public string OrderNumber { get; set; } = string.Empty;
    
    // Customer information
    [Required]
    [StringLength(255)]
    public string CustomerEmail { get; set; } = string.Empty;
    
    [StringLength(255)]
    public string? CustomerName { get; set; }
    
    // Order totals
    [Column(TypeName = "decimal(10,2)")]
    public decimal Subtotal { get; set; }
    
    [Column(TypeName = "decimal(10,2)")]
    public decimal TaxAmount { get; set; } = 0;
    
    [Column(TypeName = "decimal(10,2)")]
    public decimal ShippingAmount { get; set; } = 0;
    
    [Column(TypeName = "decimal(10,2)")]
    public decimal TotalAmount { get; set; }
    
    [Required]
    [StringLength(3)]
    public string Currency { get; set; } = "USD";
    
    // Status and metadata
    public OrderStatus Status { get; set; } = OrderStatus.Pending;

    public string? Notes { get; set; }

    // Flash sale campaign reference (optional) - SACRED schema compliant
    [Column("flash_sale_campaign_id")]
    public Guid? FlashSaleCampaignId { get; set; }

    // Navigation properties
    public FlashSale? FlashSaleCampaign { get; set; }
    public ICollection<OrderLineItem> LineItems { get; set; } = new List<OrderLineItem>();
    public ICollection<Payment> Payments { get; set; } = new List<Payment>();
}

/// <summary>
/// Individual items within an order
/// </summary>
[Table("order_line_items")]
public class OrderLineItem : BaseEntity
{
    [Required]
    public Guid OrderId { get; set; }
    
    [Required]
    public Guid SkuId { get; set; }
    
    // Item details
    [Range(1, int.MaxValue)]
    public int Quantity { get; set; }
    
    [Column(TypeName = "decimal(10,2)")]
    public decimal UnitPrice { get; set; }
    
    [Column(TypeName = "decimal(10,2)")]
    public decimal TotalPrice { get; set; }
    
    // Product snapshot (in case SKU details change)
    [Required]
    [StringLength(255)]
    public string ProductName { get; set; } = string.Empty;
    
    [Required]
    [StringLength(255)]
    public string SkuCode { get; set; } = string.Empty;
    
    // Navigation properties
    public Order Order { get; set; } = null!;
    public Sku Sku { get; set; } = null!;
}

/// <summary>
/// Payment transactions for orders
/// </summary>
[Table("payments")]
public class Payment : BaseEntity
{
    [Required]
    public Guid OrderId { get; set; }
    
    // Payment details
    [Column(TypeName = "decimal(10,2)")]
    public decimal Amount { get; set; }
    
    [Required]
    [StringLength(3)]
    public string Currency { get; set; } = "USD";
    
    [Required]
    [StringLength(50)]
    public string PaymentMethod { get; set; } = string.Empty; // credit_card, paypal, etc.
    
    // Payment gateway information
    [StringLength(255)]
    public string? GatewayTransactionId { get; set; }
    
    public string? GatewayResponse { get; set; } // JSON response from gateway
    
    // Status and metadata
    public PaymentStatus Status { get; set; } = PaymentStatus.Pending;
    
    [StringLength(100)]
    public string? ReferenceNumber { get; set; }
    
    public string? Notes { get; set; }
    
    // Navigation property
    public Order Order { get; set; } = null!;
}

File: variant-x/python-service/app/models/order.py
"""Order and transaction models."""

import uuid
from datetime import datetime
from decimal import Decimal
from enum import Enum

from sqlalchemy import Column, String, ForeignKey, DateTime, Integer, Numeric, Enum as SQLEnum, Text
from sqlalchemy.dialects.mysql import CHAR
from sqlalchemy.orm import relationship

from app.core.database import Base


class OrderStatus(str, Enum):
    """Order status enumeration."""
    pending = "pending"
    confirmed = "confirmed"
    processing = "processing"
    shipped = "shipped"
    delivered = "delivered"
    cancelled = "cancelled"
    refunded = "refunded"


class PaymentStatus(str, Enum):
    """Payment status enumeration."""
    pending = "pending"
    authorized = "authorized"
    captured = "captured"
    failed = "failed"
    cancelled = "cancelled"
    refunded = "refunded"


class Order(Base):
    """Order model for both regular and flash sale orders."""
    
    __tablename__ = "orders"
    
    id = Column(CHAR(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    order_number = Column(String(50), unique=True, nullable=False, index=True)
    
    # Customer information
    customer_email = Column(String(255), nullable=False, index=True)
    customer_name = Column(String(255), nullable=True)
    
    # Order totals
    subtotal = Column(Numeric(precision=10, scale=2), nullable=False)
    tax_amount = Column(Numeric(precision=10, scale=2), default=0, nullable=False)
    shipping_amount = Column(Numeric(precision=10, scale=2), default=0, nullable=False)
    total_amount = Column(Numeric(precision=10, scale=2), nullable=False)
    currency = Column(String(3), default="USD", nullable=False)
    
    # Status and metadata
    status = Column(SQLEnum(OrderStatus), default=OrderStatus.pending, nullable=False, index=True)
    notes = Column(Text, nullable=True)

    # Flash sale campaign reference (optional) - SACRED schema compliant
    flash_sale_campaign_id = Column(CHAR(36), ForeignKey("flash_sale_campaigns.id"), nullable=True, index=True)

    # Timestamps
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False, index=True)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

    # Relationships
    flash_sale_campaign = relationship("FlashSaleCampaign", back_populates="orders")
    line_items = relationship("OrderLineItem", back_populates="order", cascade="all, delete-orphan")
    payments = relationship("Payment", back_populates="order", cascade="all, delete-orphan")
    
    def __str__(self):
        return f"Order {self.order_number}"


class OrderLineItem(Base):
    """Individual items within an order."""
    
    __tablename__ = "order_line_items"
    
    id = Column(CHAR(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    order_id = Column(CHAR(36), ForeignKey("orders.id"), nullable=False, index=True)
    sku_id = Column(CHAR(36), ForeignKey("skus.id"), nullable=False, index=True)
    
    # Item details
    quantity = Column(Integer, nullable=False)
    unit_price = Column(Numeric(precision=10, scale=2), nullable=False)
    total_price = Column(Numeric(precision=10, scale=2), nullable=False)
    
    # Product snapshot (in case SKU details change)
    product_name = Column(String(255), nullable=False)
    sku_code = Column(String(255), nullable=False)
    
    # Timestamps
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)
    
    # Relationships
    order = relationship("Order", back_populates="line_items")
    sku = relationship("SKU")
    
    def __str__(self):
        return f"{self.quantity}x {self.product_name}"


class Payment(Base):
    """Payment transactions for orders."""
    
    __tablename__ = "payments"
    
    id = Column(CHAR(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    order_id = Column(CHAR(36), ForeignKey("orders.id"), nullable=False, index=True)
    
    # Payment details
    amount = Column(Numeric(precision=10, scale=2), nullable=False)
    currency = Column(String(3), default="USD", nullable=False)
    payment_method = Column(String(50), nullable=False)  # credit_card, paypal, etc.
    
    # Payment gateway information
    gateway_transaction_id = Column(String(255), nullable=True, index=True)
    gateway_response = Column(Text, nullable=True)  # JSON response from gateway
    
    # Status and metadata
    status = Column(SQLEnum(PaymentStatus), default=PaymentStatus.pending, nullable=False, index=True)
    reference_number = Column(String(100), nullable=True)
    notes = Column(Text, nullable=True)
    
    # Timestamps
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False, index=True)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)
    
    # Relationships
    order = relationship("Order", back_populates="payments")
    
    def __str__(self):
        return f"Payment {self.amount} {self.currency} ({self.status})"
